crates/rethnet_eth/src/remote/methods.rs

```diff
@@ -11,23 +11,66 @@ use crate::{
 /// eth_sendTransaction and eth_estimateGas
 #[derive(Clone, Debug, PartialEq, serde::Deserialize, serde::Serialize)]
 pub struct TransactionInput {
-    from: Option<Address>,
-    to: Option<Address>,
-    gas: Option<U256>,
+    /// the address from which the transaction should be sent
+    pub from: Option<Address>,
+    /// the address to which the transaction should be sent
+    pub to: Option<Address>,
+    /// gas
+    pub gas: Option<U256>,
+    /// gas price
     #[serde(rename = "gasPrice")]
-    gas_price: Option<U256>,
-    value: Option<U256>,
-    data: Option<ZeroXPrefixedBytes>,
+    pub gas_price: Option<U256>,
+    /// transaction value
+    pub value: Option<U256>,
+    /// transaction data
+    pub data: Option<ZeroXPrefixedBytes>,
 }
 
 /// for specifying the inputs to eth_newFilter
 #[derive(Clone, Debug, PartialEq, serde::Deserialize, serde::Serialize)]
 #[serde(rename_all = "camelCase")]
 pub struct FilterOptions {
-    from_block: Option<BlockSpec>,
-    to_block: Option<BlockSpec>,
-    address: Option<Address>,
-    topics: Option<Vec<ZeroXPrefixedBytes>>,
+    /// from block
+    pub from_block: Option<BlockSpec>,
+    /// to block
+    pub to_block: Option<BlockSpec>,
+    /// address
+    pub address: Option<Address>,
+    /// topics
+    pub topics: Option<Vec<ZeroXPrefixedBytes>>,
+}
+
+fn latest() -> Option<BlockSpec> {
```

Comment:
They're used for specifying the `default` attribute in the `#[serde()]` macro on the arguments to `MethodInvocation` variants that have type `Option<BlockSpec>`. As far as I can tell, only static expressions are supported for the `default` attribute. [It looks like there is a way](https://github.com/serde-rs/serde/issues/992#issuecomment-315791702) to use dynamic defaults, but this simple usage didn't seem worth the exploration of that. I'm glad you raised the question though, so that I can improve the readability. I moved these functions to be static methods of a purposefully-named struct in [this commit](https://github.com/NomicFoundation/hardhat/pull/4132/commits/86e0ab64c870a6849844fe03f0ef6ee8cfeedc15). Let me know what you think